generator/osm_source.cpp

```diff
@@ -342,56 +347,54 @@ class MainFeaturesEmitter : public EmitterBase
     static uint32_t const placeType = classif().GetTypeByPath({"place"});
     uint32_t const type = fb.GetParams().FindType(placeType, 1);
 
-    auto hotelIndex = generator::BookingDataset::kInvalidHotelIndex;
-
+    // TODO(mgserigio): Would it be better to have objects that store callback
+    // and can be piped: action-if-cond1 | action-if-cond-2 | ... ?
+    // The first object which perform action terminates the cahin.
```

Comment:
s/cahin/chain

Short answer to your question: yes, it would. But co-routines must be implemented first.